generators/server/templates/src/main/java/package/repository/_UserRepository.java

```diff
@@ -80,7 +82,12 @@
 
     Optional<User> findOneByResetKey(String resetKey);
 
+    <%_ if (databaseType === 'sql') { _%>
+    @Query("from User u where lower(u.email) = lower(:email)")
+    Optional<User> findOneByEmail(@Param("email") String email);
```

Comment:
This can be done without an annotation by using the JPA query `findOneByEmailIgnoreCase`

That would work for Mongo but I'm not really sure how Cassandra works

Reply:
I didn't want to change the method name conditionally based on databaseType - it is called from more than 20 places - it would mean a lot's of if's.

Reply:
why not use `findOneByEmailIgnoreCase` as the method name for all DB?

Reply:
and for cassandra since its a method hand written we could do something like
```
    public Optional<User> findOneByEmailIgnoreCase(String email) {
        BoundStatement stmt = findOneByEmailStmt.bind();
        stmt.setString("email", email);
        return findOneFromIndex(stmt);
    }
```
and in the save method we can convert the email to lowercase